src/Sentry.Unity.Editor.iOS/NativeMain.cs

```diff
@@ -0,0 +1,80 @@
+using System.IO;
+using System.Text.RegularExpressions;
+using UnityEngine;
+
+namespace Sentry.Unity.Editor.iOS
+{
+    internal interface INativeMain
+    {
+        public void AddSentry(string pathToMain);
+    }
+
+    internal class NativeMain : INativeMain
+    {
+        private const string Include = @"#include <Sentry/Sentry.h>
+#include ""SentryOptions.m""
+";
+        private const string Init = @"
+        [SentrySDK startWithOptions:getSentryOptions()];
+";
+
+        public void AddSentry(string pathToMain)
+        {
+            if (!DoesMainExist(pathToMain))
+            {
+                return;
+            }
+
+            var main = File.ReadAllText(pathToMain);
+            if (ContainsSentry(main))
+            {
+                return;
+            }
+
+            var sentryMain = AddSentryToMain(main);
+            if (sentryMain is null)
+            {
+                return;
+            }
+
+            File.WriteAllText(pathToMain, sentryMain);
```

Comment:
There are 3 exit conditions before we add Sentry. It could be useful when troubleshooting to log Debug (or error on `AddSentryToMain` returning null?) to help us understand customers issues when they report the logs